nbtc/relayer.go

```diff
@@ -0,0 +1,134 @@
+package nbtc
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/gonative-cc/relayer/bitcoin"
+	"github.com/gonative-cc/relayer/dal"
+	"github.com/rs/zerolog/log"
+)
+
+// Relayer broadcasts pending transactions from the database to the Bitcoin network.
+type Relayer struct {
+	db           *dal.DB
+	btcClient    bitcoin.Client
+	shutdownChan chan struct{}
+}
+
+// NewRelayer creates a new Relayer instance with the given configuration.
+func NewRelayer(db *dal.DB) (*Relayer, error) {
+	if db == nil {
+		return nil, fmt.Errorf("database cannot be nil")
+	}
+
+	host := os.Getenv("BTC_RPC")
+	user := os.Getenv("BTC_RPC_USER")
+	pass := os.Getenv("BTC_RPC_PASS")
```

Comment:
Earlier you mentioned to check if the config values like the BTC_RPC are non-empty. Do we keep the check in the realyer since now the config is provided by the caller? I am not sure what is the best approach